compiler/prelude/prelude.go

```diff
@@ -120,7 +120,7 @@ var $substring = function(str, low, high) {
 
 var $sliceToArray = function(slice) {
   if (slice.$length === 0) {
-    return [];
+    return new slice.$array.constructor;
```

Comment:
Can we simply remove this special case? `subarray` and `slice` with a length of `0` should give the same result.